packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/fixtures/types-based-on-inheritance.ts.input

```diff
@@ -0,0 +1,63 @@
+class Animal {
```

Comment:
Can you add cases for:


Multiple Implements
```
interface Thing {
  thing(num: number): void
}

class MultipleImplements implements Wild, Thing {
  wild(place) { return 'foo' };
  thing(num) { console.log(num); };
}
```

Generic Interfaces
```
interface GenericThing<T = number> {
  thing(num: T): void
}

type Thing = { foo: number };

interface ConstrainedThing<T extends Thing> {
  thing(num: T): void
}

class GenericImpl implements GenericThing {
  thing(num) { console.log(num); };
}

class GenericImplTwo implements GenericThing<number> {
  thing(num) { console.log(num); };
}

class GenericImplTwo implements ConstrainedThing<{ foo: number, bar: string }> {
  thing(num) { console.log(num); };
}
```

Generic Classes
```
class GenericThing<T = number> {
  thing(num: T): void
}

type Thing = { foo: number };

class ConstrainedThing<T extends Thing> {
  thing(num: T) {
  
  }
}

class GenericImpl extends GenericThing {
  thing(num) { console.log(num); };
}

class GenericImplTwo extends GenericThing<number> {
  thing(num) { console.log(num); };
}

class GenericImplTwo extends ConstrainedThing<{ foo: number, bar: string }> {
  thing(num) { console.log(num); };
}
```

Reply:
Test cases added and handled :)